src/eval_context.rs

```diff
@@ -872,7 +871,8 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
         let pointee_size = self.type_size(pointee_ty)?.expect("cannot offset a pointer to an unsized type") as i64;
         return if let Some(offset) = offset.checked_mul(pointee_size) {
             let ptr = ptr.signed_offset(offset, self.memory.layout)?;
-            self.memory.check_bounds(ptr, false)?;
+            // Why are we doing a bounds check on a pointer?
```

Comment:
I think so. The standard library uses `arith_offset` when using integer pointers.